### PR TITLE
[4.2] Improving javascript validation of package installer

### PR DESCRIPTION
**Pull request by vorayash**
Pull Request for Issue #37575.

### Summary of Changes
Added validation for non zip files in 'build\media_source\plg_installer_packageinstaller\js\packageinstaller.es6.js'

### Testing Instructions
Try to upload an extension in the "upload package file".

### Actual result BEFORE applying this Pull Request
The error was showing just red colored error bar without any content.
image

### Expected result AFTER applying this Pull Request
image

### CODE REVIEW

**Comment by laoneo** [comment]
I would more ask why is there no message in the alert. I think it would be better to serve a proper message from the server why the installation failed. Did you figure out where this empty error comes from?

File: build/media_source/plg_installer_packageinstaller/js/packageinstaller.es6.js [hunk 1]
```diff
@@ -125,6 +125,13 @@ Joomla = window.Joomla || {};
         return;
       }
 
+      if (file.name.split('.').pop() !== "zip") {
```
Comment (dgrammatiko) [inline build/media_source/plg_installer_packageinstaller/js/packageinstaller.es6.js]: You shouldn't really rely on the extension (for security reasons) on any file check in the web. A better approach is to check the mime type, something like: `file.type !== 'application/zip'`. In sort the extension should NEVER be used to sniff the type of file

**Review by dgrammatiko — CHANGES_REQUESTED** [review]

**Comment by vorayash** [comment]
> I would more ask why is there no message in the alert. I think it would be better to serve a proper message from the server why the installation failed. Did you figure out where this empty error comes from?

When we send folder as data, xhr readystate changes to 4 and status to 0. It means request is failed. When request fails, value of xhr.statusText changes into empty string(""). 

image

xhr.statusText is a string so it will satisfy below-pointed condition and empty message will be shown.

image

I tried xhr request on my local machine and found that when we send folder as data, request fails. So I think sending folder is not allowed in xmlhttprequest.

File: build/media_source/plg_installer_packageinstaller/js/packageinstaller.es6.js (hunk 1 above)
Comment (vorayash) [reply build/media_source/plg_installer_packageinstaller/js/packageinstaller.es6.js]: 👍

**Comment by laoneo** [comment]
So it means the request never arrives on the server?

**Comment by vorayash** [comment]
I think so. Can you please confirm that?

**Comment by Fedik** [comment]
It should be fixed the empty message, instead. 
Server already doing validation. as @laoneo pointed.

**Pull request closed without merging**